### PR TITLE
grt: update saveGuides and load guides from odb

### PR DESCRIPTION
**Pull request by eder-matheus**
Fixes https://github.com/The-OpenROAD-Project/OpenROAD/issues/2399

### CODE REVIEW

**Comment by maliberty** [comment]
Did you mean to add these:
image
?

**Comment by eder-matheus** [comment]
> Did you mean to add these: image ?

Only the `.odb`, I just removed the `.db`. Is this file too large to be in a unit test?

**Comment by maliberty** [comment]
> > Did you mean to add these: image ?
> 
> Only the `.odb`, I just removed the `.db`. Is this file too large to be in a unit test?

I'm more concerned that the .odb will be problem if we have a schema update.  Can a .def be used here (possibly gzipped)?

**Comment by eder-matheus** [comment]
> > > Did you mean to add these: image ?
> > 
> > 
> > Only the `.odb`, I just removed the `.db`. Is this file too large to be in a unit test?
> 
> I'm more concerned that the .odb will be problem if we have a schema update. Can a .def be used here (possibly gzipped)?

The `est_rc3` already covers reading the def and the guide file. I think I can just remove the `est_rc4 `at all.

**Comment by eder-matheus** [comment]
I have these failures on secure-ci: 
* gf12/ariane: `[ERROR] finish__timing__setup__ws fail test`
* gf12/bp_single: `[ERROR] detailedroute__route__drc_errors fail test: 4.0 <= 0.0`
* gf12/swerv_wrapper: `[ERROR] finish__timing__drv__max_slew_limit fail test`
But I think they're the same as that we have on the master branch.

gf12/ca53 is passing with my updates. The differences start at CTS, so I don't think they're related to my changes. I think it is safe to merge this update.

**Comment by maliberty** [comment]
[multiple_calls.ok](https://github.com/The-OpenROAD-Project/OpenROAD/pull/2539/files/fb5a12edcf2aeca4f4242473020431ce4fe50947#diff-a6c247c2527a6db3be61ca5caf20adfff8c2a161bb40c3a94cab8958ee9945c4) is still not resolved

**Review by maliberty — COMMENTED** [review]
Sorry I see this was still pending

File: src/grt/test/multiple_calls.ok [hunk 1]
```diff
@@ -129,5 +129,9 @@ Total            31824           655            2.06%             0 /  0 /  0
...
 [INFO GRT-0018] Total wirelength: 2347 um
 [INFO GRT-0014] Routed nets: 87
-No differences found.
-No differences found.
+Differences found at line 4.
+58800 105000 63000 113400 metal2
+58800 105000 63000 109200 metal1
+Differences found at line 4.
+58800 105000 63000 113400 metal2
+58800 105000 63000 109200 metal1
```
[the remark this answers is not in the document]
Comment (eder-matheus) [reply src/grt/test/multiple_calls.ok]: Thanks, I didn't notice this before. Fixed now.

**Review by maliberty — APPROVED** [review]

**Pull request merged**